src/onemod/fsutils/data_loader.py

```diff
@@ -0,0 +1,77 @@
+from pathlib import Path
+from typing import Literal
+
+import pandas as pd
+import polars as pl
+
+from onemod.fsutils.io import DataIO, dataio_dict
+
+
+class DataLoader:
+    """Handles loading and dumping of data files with optional lazy loading,
+    column selection, and subset filtering."""
+
+    io_dict: dict[str, DataIO] = dataio_dict
+
+    def load(
+        self,
+        path: Path,
+        lazy: bool = False,
+        backend: Literal["polars", "pandas"] = "polars",
+        columns: list[str] | None = None,
+        id_subsets: dict[str, list] | None = None,
+        **options,
+    ) -> pl.DataFrame | pl.LazyFrame | pd.DataFrame:
+        """Load data with optional lazy loading and subset filtering. Supports
+        both Polars and Pandas backends."""
+        if lazy and backend == "pandas":
+            raise ValueError("Pandas backend does not support lazy loading")
+
+        if path.suffix not in self.io_dict:
+            raise ValueError(f"Unsupported data format for '{path.suffix}'")
+
+        if backend == "polars":
```

Comment:
Here we can always use the lazy frame to load and if `"polars"` eager we `collect` and if `"pandas"` we collect and convert to pandas data frame :)

Reply:
This is an interesting proposition. It sounds like something I would want to do in order to simplify code and make maintaining this logic easier, esp if we end up expanding `DataInterface` and `load()` responsibilities in the future, which seems likely. So there are certainly potential pros. In terms of memory and performance, I expect it would usually be a net improvement, but I actually bet it depends. There is non-zero overhead in a to_pandas conversion step.

More importantly, I think this would be fundamentally changing what is meant by "backend", to the point that we might even call it something else. Admittedly I have already pretty much done this with `"polars_lazy"` vs `"polars_eager"`.. so the way I have done that is actually questionable in the opposite direction I think.

But what I mean is I think this proposition would not come without tradeoffs. I added the `pandas` logic separately because I imagine if someone (let's say they are unfamiliar with polars) specified `"pandas"` for `backend` and they ran into loading errors (whether at reading, subsetting, whichever stage), they would expect the errors they receive to be the Pandas errors they are familiar with, right? Whereas if behind the scenes we are actually still using LazyFrames, the issues they might run into could be very confusing for them (like "I explicitly chose Pandas backend, so why am I getting Polars errors?"). 
Now, we could change the argument name to "return_type" or something, to make it less misleading, but then there's no option for pandas backend at all. This is not something that I personally would care about, since I don't think I would personally use pandas for most big data tasks and I am comfortable receiving polars errors and warnings from my load function, so I will defer to you and others on what seems the most useful and intuitive from the average user's perspective

Reply:
Thanks very much for being so thoughtful @blsmxiu47! And it is a good point, indeed if we change the behavior to always use `polar_lazy`, `backend="pandas"` might contradict with what user expect. I like the idea of changing it to `return_type` and always sticking with `backend=polar_lazy`.

Reply:
Updated to this approach. Code does look cleaner now, thanks! But let us know if there is ever a demand for pandas (or other) backend.

Left the argument as a string literal. Not sure if it would be better practice to actually require a Type to be specified for `return_type` (e.g. `pl.DataFrame` instead of `"polars_dataframe"`). Either way works, but the non-string method requires the library to be imported for instance (which it probably would be, but still..) Anyway want to do whatever is easiest and most intuitive from the users' POV. lmk if you want me to change it.

Reply:
This looks good! Thanks very much!